pkgs/applications/networking/portmaster/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+}:
+
+buildGoModule rec {
+  pname = "portmaster";
+  version = "1.5.1";
+
+  src = fetchFromGitHub {
+    owner = "safing";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-XlFPM6EaoSGOV39aZW6fG9Quacw7Z4lSxDeSW2excL0=";
+  };
+
+  vendorHash = "sha256-dj+XDrcbZXdIj1WlTb5Y+v88by54t3R8zLujjtKiudA=";
+
+  # integration tests require root access
+  doCheck = false;
```

Comment:
Could they be implemented as nixos tests?

Reply:
Not sure, pretty new to packaging in general, at it current state it just compiles and doesn't even start yet.

Reply:
> Not sure, pretty new to packaging in general, at it current state it just compiles and doesn't even start yet.

Before implementing a NixOS test, there should be a NixOS module... How do you plan to use this package @Nyabinary ?

Reply:
> > Not sure, pretty new to packaging in general, at it current state it just compiles and doesn't even start yet.
> 
> Before implementing a NixOS test, there should be a NixOS module... How do you plan to use this package @Nyabinary ?

Portmaster is a firewall + vpn service + network activity viewer rolled into one, so that how I intend to use it

Reply:
> so that how I intend to use it

What I meant was, how do you intend to run it in the background? Using systemd service you'd define in your NixOS configuration?

Reply:
Upstream provides a systemd service file :3